### PR TITLE
docs: update addressing with ipfs uri

### PR DESCRIPTION
**Pull request by olizilla**
- replace references to "native url" with "ipfs uri" to be consistent with https://docs.ipfs.io/how-to/best-practices-for-nft-data/#types-of-ipfs-links-and-when-to-use-them
- use dweb.link consistently for http-to-ipfs gateway urls
- simplify the TL;DR

### CODE REVIEW

File: ADDRESSING.md [hunk 1]
```diff
@@ -80,13 +83,13 @@ Outside of browser context, and in cases when site isolation does not matter (se
 
 Examples:
 
-    https://gateway.ipfs.io/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/wiki/Vincent_van_Gogh.html
-    https://gateway.ipfs.io/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
-    https://gateway.ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
+    https://dweb.link/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/wiki/Vincent_van_Gogh.html
+    https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
+    https://dweb.link/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
```
Comment (lidel) [inline ADDRESSING.md:88]: The dnslink is mutable, recently changed to version without `.html` suffix, and may change over time, so better to use generic landing page in this example:
```suggestion
    https://dweb.link/ipns/tr.wikipedia-on-ipfs.org/wiki/
```

File: ADDRESSING.md [hunk 2]
```diff
@@ -20,16 +18,21 @@
...
-- mutable content behind human-readable DNSLink (DNS TXT record): `ipns://{fqdn}` 
+For immutable content use an IPFS URI (`ipfs://{cid}`).
+- [ipfs://bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy](ipfs://bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy)
+
+For mutable content use an IPNS DNSLink URI (`ipns://{fqdn}`).
+- [ipns://webui.ipfs.io](ipns://webui.ipfs.io)
+
+For web browsers and other tools that do not yet recognise the `ipfs://` and `ipns://` schemes, additionally provide an HTTP-to-IPFS gateway URL.
+- [https://dweb.link/ipfs/bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy](https://dweb.link/ipfs/bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy)
+- [https://dweb.link/ipns/webui.ipfs.io](https://dweb.link/ipns/webui.ipfs.io)
```
Comment (lidel) [inline ADDRESSING.md:29]: Webui requires CORS so would show "error" by default for new users, this may be a better example for demo purposes:
```suggestion
- [https://dweb.link/ipns/cid.ipfs.io](https://dweb.link/ipns/cid.ipfs.io)
```

File: ADDRESSING.md [hunk 3]
```diff
@@ -20,16 +18,21 @@
...
 
-Native URI can be used for addressing:
-- immutable content:  `ipfs://{cid}`
-- mutable content behind IPNS record signed with a specific libp2p key: `ipns://{libp2p-key}` 
-- mutable content behind human-readable DNSLink (DNS TXT record): `ipns://{fqdn}` 
+For immutable content use an IPFS URI (`ipfs://{cid}`).
+- [ipfs://bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy](ipfs://bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy)
+
+For mutable content use an IPNS DNSLink URI (`ipns://{fqdn}`).
+- [ipns://webui.ipfs.io](ipns://webui.ipfs.io)
```
Comment (lidel) [inline ADDRESSING.md:25]:
```suggestion
- [ipns://cid.ipfs.io](ipns://cid.ipfs.io)
- [ipns://k51qzi5uqu5dgutdk6i1ynyzgkqngpha5xpgia3a5qqp4jsh0u4csozksxel2r](ipns://k51qzi5uqu5dgutdk6i1ynyzgkqngpha5xpgia3a5qqp4jsh0u4csozksxel2r)
```

I think it is better if we add it, even if this is this silly example for now.
The risk of cryptographic identifiers being ignored in implementations is too high.

File: ADDRESSING.md [hunk 4]
```diff
@@ -20,16 +18,21 @@
...
 ## TL;DR
 
-Native URI can be used for addressing:
-- immutable content:  `ipfs://{cid}`
-- mutable content behind IPNS record signed with a specific libp2p key: `ipns://{libp2p-key}` 
-- mutable content behind human-readable DNSLink (DNS TXT record): `ipns://{fqdn}` 
+For immutable content use an IPFS URI (`ipfs://{cid}`).
+- [ipfs://bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy](ipfs://bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy)
+
+For mutable content use an IPNS DNSLink URI (`ipns://{fqdn}`).
```
Comment (lidel) [inline ADDRESSING.md:24]: Not mentioning `{libp2p-key}` here is a problem – people will cur corners and hardcode `ipns://` validation as "oh, is this a valid FQDN".  At the very least we should:

```suggestion
For mutable content use an IPNS DNSLink URI (human-readable `ipns://{fqdn}` or cryptographic  `ipns://{libp2p-key}`).
```

**Review by lidel — CHANGES_REQUESTED** [review]
@olizilla looks good, but we need to ensure cryptographic `ipns://` is included. 
Mind addressing my comments inline?

File: ADDRESSING.md (hunk 4 above)
Comment (olizilla) [reply ADDRESSING.md:24]: I was shooting for have the TL;DR make a clear recommendation, and letting the rest of the doc explain all the things. I'm not adovcating for removing ipns://{libp2p-key} from the doc, just streamlining the TL;DR.

**Review by lidel — APPROVED** [review]
LGTM, thank you @olizilla :heart: 
I pushed a small tweak with replaces "fqdn" with simple "dnslink". 
Merging this, but feel free to tweak things further in separate PRs if needed.
